src/Interactive/Features/Interactive/Core/InteractiveHost.cs

```diff
@@ -231,21 +232,24 @@ private bool CheckAlive(Process process)
 
         public void Dispose()
         {
-            Dispose(joinThreads: false, disposing: true);
+            Dispose(joinThreads: true, disposing: true);
         }
 
         private void Dispose(bool joinThreads, bool disposing)
```

Comment:
>bool joinThreads, bool disposing [](start = 29, length = 32)

I think we should remove joinThreads and use disposing instead on line 250. We don't want to be waiting during finalization (on threads to join).